### PR TITLE
Hand of God

### PR DESCRIPTION
**Pull request by XeonFaux**
Ports over https://github.com/tgstation/-tg-station/pull/11803 as well as https://github.com/tgstation/-tg-station/pull/13080

### CODE REVIEW

**Comment by doobiedoo23** [comment]
Might be a good idea to add a description for people who don't know what HoG is

File: code/__DEFINES/role_preferences.dm [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	ROLE_ALIEN,
+	ROLE_PAI,
+	ROLE_CULTIST = /datum/game_mode/cult,
+	ROLE_BLOB = /datum/game_mode/blob,
+	ROLE_NINJA,
+	ROLE_MONKEY = /datum/game_mode/monkey,
+	ROLE_GANG = /datum/game_mode/gang,
+	ROLE_SHADOWLING = /datum/game_mode/shadowling,
+	ROLE_ABDUCTOR = /datum/game_mode/abduction,
+	ROLE_HOG_GOD = /datum/game_mode/hand_of_god,
```
Comment (danhibiki) [inline code/__DEFINES/role_preferences.dm:48]: Revenant should be your next defined special_roles according to your defines, no?

Comment (XeonFaux) [inline code/__DEFINES/role_preferences.dm:48]: Ah, I see I forgot that one.

**Comment by TDcoolguy300** [comment]
This does modify the controllers, albeit small.

**Comment by nalarmg** [comment]
Needs a proper description, or at least a link to where one can be found if this is a port.

**Comment by Nokington** [comment]
I don't even know what this is

File: html/changelogs/XeonFaux - HOG.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+author: XeonFaux
+
+delete-after: True
+
+changes: 
+  - rscadd: "Hand of God: Coming to a gamemode near you!"
```
Comment (nalarmg) [inline html/changelogs/XeonFaux - HOG.yml]: Very undescriptive and unhelpful. What's the gamemode about, how will this effect existing mechanics, when will it be playable. Is this just a framework that's unplayable - if so, why does this matter to the players? Changelogs should ideally be informative and let the players know exactly as to what's changed and how it effects them.

**Comment by Nokington** [comment]
Will this be in the rotation when merged?

**Comment by nalarmg** [comment]
As that's a config thing, it'd depend on whether or not the heads want it, and whether or not myself/admiral hippie's informed of the decision.

**Comment by Nokington** [comment]
Alright, i'd rather wait to put it in rotation and have a couple test rounds with it. so :+1: for now

**Comment by Lawligagger** [comment]
Yeah, leave it out of rotation so we can do a few test rounds with it.

**Comment by Nokington** [comment]
Merge conflicts yo

**Comment by blobbernaut** [comment]
what the hell is hand of god

**Comment by XeonFaux** [comment]
Read the changelog and/or the links given in original description.

**Comment by FuryMcFlurry** [comment]
my attempt on making the armor looks a littler better.
image

File: code/__DEFINES/role_preferences.dm [hunk 3]
```diff
@@ -0,0 +1,51 @@
+
+
+//Values for antag preferences, event roles, etc. unified here
+
+
+
+//These are synced with the Database, if you change the values of the defines
+//then you MUST update the database!
```
Comment (Carbonhell) [inline code/__DEFINES/role_preferences.dm:8]: Does this need any sort of edit to our database..?

File: code/game/gamemodes/handofgod/items.dm [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+/obj/item/weapon/storage/backpack/bannerpack/blue
+	name = "blue banner backpack"
+	desc = "It's a backpack with lots of extra room.  A blue banner is attached, that can't be removed."
+	icon_state = "bannerpack-blue"
+
+
+
+//this is all part of one item set
+/obj/item/clothing/suit/armor/plate/crusader
+	name = "Crusader's Armour"
```
Comment (Carbonhell) [inline code/game/gamemodes/handofgod/items.dm]: might be wise to rename this to differentiate it from chaplain's crusade armor

File: code/game/gamemodes/handofgod/items.dm [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+/obj/item/clothing/suit/armor/plate/crusader/examine(mob/user)
+	..()
+	if(!is_handofgod_cultist(user))
+		user << "Armour that's comprised of metal and cloth."
+	else
+		user << "Armour that was used to protect from backstabs, gunshots, explosives, and lasers.  The original wearers of this type of armour were trying to avoid being murdered.  Since they're not around anymore, you're not sure if they were successful or not."
+
+
+/obj/item/clothing/head/helmet/plate/crusader
+	name = "Crusader's Hood"
```
Comment (Carbonhell) [inline code/game/gamemodes/handofgod/items.dm]: same here

File: code/game/gamemodes/handofgod/powers.dm [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+
+			if(success)
+				H << "<span class='boldnotice'>A powerful staff has been bestowed upon you, you can use this to convert the false god's structures!</span>"
+				H << "<span class=boldnotice'>[success]</span>"
+		//end prophet gear
+
+		add_faith(-100)
+
+
+/mob/camera/god/verb/talk(msg as text)
```
Comment (Carbonhell) [inline code/game/gamemodes/handofgod/powers.dm:161]: i don't see anything that logs this in the say logs, that should be added(should be log_say or something)

File: code/game/gamemodes/handofgod/powers.dm [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+	set desc = "Instantly creates your nexus, You can only do this once, make sure you're happy with it!"
+
+	if(!ability_cost(0,1,0))
+		return
+
+	place_nexus()
+
+
+/* //Transolocators have no sprite
+/mob/camera/god/verb/movenexus()
```
Comment (Carbonhell) [inline code/game/gamemodes/handofgod/powers.dm:226]: If the sprite's the only problem just ask fury to make one, code seems fine?

File: code/game/gamemodes/handofgod/structures.dm [hunk 8]
```diff
@@ -0,0 +1,591 @@
...
+	glass_cost = 5
+
+/obj/structure/divine/conduit/assign_deity(mob/camera/god/new_deity, alert_old_deity = TRUE)
+	if(deity)
+		deity.conduits -= src
+	..()
+	if(deity)
+		deity.conduits += src
+
+/* //No good sprites, and not enough items to make it viable yet
```
Comment (Carbonhell) [inline code/game/gamemodes/handofgod/structures.dm:282]: "No good sprites" we can easily fix that, and for items,it shouldn't be too hard!

File: code/game/gamemodes/handofgod/structures.dm [hunk 9]
```diff
@@ -0,0 +1,591 @@
...
+
+/obj/structure/divine/shrine/assign_deity(mob/camera/god/new_deity, alert_old_deity = TRUE)
+	if(..())
+		name = "shrine to [new_deity.name]"
+		desc = "A shrine dedicated to [new_deity.name]"
+
+
+
+
+//Functional, but need sprites
```
Comment (Carbonhell) [inline code/game/gamemodes/handofgod/structures.dm:549]: "//Functional, but need sprites" @FuryMcFlurry AAAAAAAA

File: code/modules/client/preferences_savefile.dm [hunk 10]
```diff
@@ -30,6 +30,67 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 		return savefile_version
 	return -1
 
+/datum/preferences/proc/update_antagchoices(current_version)
```
Comment (Carbonhell) [inline code/modules/client/preferences_savefile.dm:33]: Does this need a preferences savefile update or not?

**Comment by Carbonhell** [comment]
Code looks good,i've checked all of it, it should work. Through if possible it'd be better to finish it since most of the commented out features require sprites, and we have a pretty good spriter,so they won't end forgotten in some random dm file forever

**Comment by Nokington** [comment]
Thank you @Carbonhell for the tons of emails about comments in this PRs.

**Comment by danhibiki** [comment]
I'd actually like to genuinely thank carbon for going thru the PR so closely. Kudos man

**Comment by Nokington** [comment]
Yeah good job Carbon but now I know to turn off notifications.

**Comment by XeonFaux** [comment]
Actually, I believe you ARE going to need to modify the database, for the antag list. Make sure to back it up if this does go through.

File: code/__DEFINES/role_preferences.dm (hunk 3 above)
Comment (XeonFaux) [inline code/__DEFINES/role_preferences.dm:8]: No

**Comment by FuryMcFlurry** [comment]
convertion
convertion altar, shitty gif but the animation loops just fine

**Comment by Nokington** [comment]
:+1: Still as long as it is not in rotation.

**Comment by nalarmg** [comment]
Is this suitable/ready for merge?

**Comment by XeonFaux** [comment]
This is ready for merge.

**Pull request merged**